### ISSUE TITLE
Github Actions: error when reading local file

### ISSUE THREAD

**Issue by nilodna** [oceanhackweek/OHW22_SA_ocean_db#16]
To use the github-actions for a serverless deployment of the data gathering part of our project, we must change how the code @douglasnehme wrote operates. Instead of saving a .csv file every time we run the `fetch_pnboia.py`, this function might return a `DataFrame` to the `pipeline.py` code. The same method was applied for quality control. 

So, to summarize, what I'm thinking about implementing is something like that:

On the `fetch_pnboia.py` file:
```
def pnboia_update_local_data(buoy_id):
    # perform the download
    return df
```

So the `pipeline.py` file will work like this:
```
for buoy in buoys:
    # Updating local file with the last data from PNBOIA
    df = pnboia_update_local_data(buoy_id=buoy)
    
    # applying quality control pipeline for each active PNBOIA
    df = run_qc(df)

    # sending data to the database
    send_data_to_database(df, tablename='pnboia')
```

With this new workflow, we can use the github-actions to run this pipeline hourly to save the data into our database. With this method, we can also avoid consuming too much space on our Github repository.

I'll put this in action maybe this week, on a separate branch, to test and check if everything runs ok.

**Comment by douglasnehme** [comment]
Good idea, this will save space and time. But I can't do it this week, because I have a lot of commitments from my PhD. Once this is implemented I say here.

**Comment by nilodna** [comment]
Don't worry @douglasnehme. I was planning to do this myself but if you have an interest in coding this adjustment, we can discuss it better.

**Comment by douglasnehme** [comment]
I planned to code this too, @nilodna. But if august 28 is a late deadline for any reason you can do it.

**Comment by nilodna** [comment]
No rush, @douglasnehme. I'll let this issue for you, but if you need any assistance, let me know.

**Comment by douglasnehme** [comment]
@nilodna, I started to code this issue in a new branch on my local rep and came across a doubt.

The `pnboia_update_local_data` function in `fetch_pnboia.py` before to create or update a .csv file every time it is run it try to read an existing .csv. In the new workflow this function will return a DataFrame instead of save/update any file and consequently will stop to read anything too.

Here is my doubt, because if we want to continually ingest new data to our database this function need to read it from the database. If i am right how can I do it?

**Comment by nilodna** [comment]
Hi @douglasnehme 

I believe you need to read the information from the database to avoid duplicates, right? In this case, you don't need to worry, because in the final step (send the data to the database) I coded a function that drops any duplicated row , comparing the DataFrame and the Database. 

Check this function [here](https://github.com/oceanhackweek/OHW22_SA_ocean_db/blob/main/ohw22sadb/database.py#L30) to see how it operates. But basically, what I'm doing is just using the `datetime` and `buoy` columns from the DataFrame and deleting any row that matches with both conditions. This way we avoid duplicates in the database and update any wrong values sent in the last run of our pipeline.

This information solves your doubt?

**Comment by douglasnehme** [comment]
Thank you, @nilodna. It solve my doubt.
